lib/nylas/http_client.rb

```diff
@@ -86,14 +86,12 @@ def execute(method:, path: nil, headers: {}, query: {}, payload: nil)
         timeout: timeout
       )
       rest_client_execute(**request) do |response, _request, result|
-        content_type = nil
-
-        if response.headers && response.headers[:content_type]
-          content_type = response.headers[:content_type].downcase
+        # Try to parse all responses as JSON, since api.nylas.com content-type can be flaky
+        begin
+          response = parse_response(response)
+        rescue Nylas::JsonParseError
```

Comment:
@ozsivanov Should we assume that all `Nylas::JsonParseError` are non-JSON responses (files) and therefore pass them on?  I think there would be a case where an exception is raised due to bad JSON.  Is there a way to catch that and not rescue the exception in that case?

Reply:
I don't think there is a way for us to distinguish between those two cases. I'd like to think that api.nylas.com won't respond with malformed JSON, since we would be currently seeing `Nylas::JsonParseError` getting thrown.